src/test/unit_tests/test_encryption.js

```diff
@@ -0,0 +1,81 @@
+/* Copyright (C) 2016 NooBaa */
```

Comment:
I think that the addition of regenerating access keys for accounts is also a good test case.
Edit cloud connections and see the regeneration.
Check the hierarchy on system_store load (that all master_keys are resolved in the cache of master_keys_manager).

Reply:
Addad many tests, please review it again 👍